src/components/modals/index.tsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+
+import { ButtonWithModal } from './ButtonWithModal';
+import { ActionButton } from './ModalWithActionButtons';
+
+export interface ButtonConfirmationModalProps {
+  modalTitle: string;
+  modalContentText: string;
+  name: string;
+  confirmButtonProps: ActionButton;
+  cancelButtonProps: ActionButton;
+}
+
+export const ButtonWithConfirmationModal = ({
```

Comment:
for this one, I think the interface should be that the props for the buttons are optional, then we tag whatever is passed in as onClick to the confirmButton.

that way user only uses confirm/cancel buttonprops for styling and doesn't have to worry about passing in weird props

a good way of thinking about this is that this is a button, so if i want a button to execute something when I click it i pass in a function to its onclick prop.

but now i have to pass in that function to confirmButtonProps

Reply:
This only applies to ButtonWithConfirmationModal right? Otherwise, if ppl wanted 2+ buttons with custom onClicks then they can't do that if we're going with what you're proposing here.

Reply:
Event then, this means I'd have to change ButtonWithModal if I change ButtonWithConfirmationModal based on what's listed here

Reply:
well i mean you could still pass through onClick to the cancelButtonProp,  but if i want a button that does something just with a modal i should able to pass that something into onClick

I think its fine for now - modals shouldn't have more than 2 buttons from a design standpoint

Reply:
So I go with what you proposed then? Just wanted to clarify when u said "it's fine for now"

Reply:
@godwinpang

Reply:
yeah